site/en/android-environment/lint-rules-for-ChromeOS-in-Android-Studio.md

```diff
@@ -0,0 +1,165 @@
+lint rules provide developers with more visibility into how their apps run on
+ChromeOS, showcasing issues, both software and hardware, that will undoubtedly
+cause issues for Android applications that are run on any ChromeOS device.
+
+To get more background context on the existence of these lint rules and their
+importance, read through our blog post.
+
+# Where are these lint rules?
+
+We have been in active development for a few months now. With the release
+schedule of Android Studio, some lint rules are being introduced with the
+Electric Eel Canary builds. A few of these lint rules are also now available in
+the Flamingo Canary releases. We will continue to build towards having these in
+the stable versions of Android Studio in the coming months.
+
+Another major thing to note is that these rules will be ***enabled by
+default*** in newer versions of Android Studio. The goal of this is to have more
+strongly opinionated guidance on how we want to help engineers build for
+ChromeOS and larger screens going forward.
+
+# New lint rules (updated as of Flamingo Canary 3)
+
+### x86/x86_64 ABI support
+
+The majority of Chromebooks run on the Intel architecture, thus making them a
+predominantly x86 architectural platform. For ChromeOS to properly be supported
+when NDK code is included as part of the binary, having x86 is a performance
+boost due to removing the translation required from ARM libraries. Therefore, it
+is strongly recommended that your development team adds `x86` or preferably
+`x86_64` architecture support as it would be a performance boon for any native
+code on ChromeOS or any Intel device.
+
+#### Remedy
+
+If possible, add `x86` and `x86_64` inside of your `abiSplits` within your
+`build.gradle`. Also, ensure that you are adding the code to the appropriate
+folders to support these ABIs. To get more information, reference the
+documentation on [Android ABIs](https://developer.android.com/ndk/guides/abis)
+and the talk on [ABI Support from ADS](https://youtu.be/C0IuT0O2wlM?t=229).
+
+!!! aside.message--note
+**Note:** Ensure that whatever included third-party libraries that are being
+utilized also have x86 and x86_64 binaries. 
+!!!
+
+### ChromeOS hardware limitation
+
+The majority of ChromeOS devices come with a smaller sample set of hardware
+sensors and other features compared to an Android phone. The goal of this rule
+is to flag to developers that if you are using the flag `<uses-feature>` with
+`android:required=true`, your app will not be available on the Google Play Store
+on ChromeOS. A strong recommendation to ensure that your app can be accessed on
+as many devices as possible is to ensure that the hardware feature is not
+required by default. Instead, you can add defensive code to check for specific
+hardware at run time rather. An example of this would be
+
+```
+<uses-feature android:name="android.hardware.camera" android:required="true">
```

Comment:
```suggestion
%lt;uses-feature android:name="android.hardware.camera" android:required="true"%gt;
```

Ok, I _think_ this is why builds are failing. Because the `<>` aren't encoded, they're trying to be parsed as HTML and then failing because it's invalid HTML. This should encode them and all should be OK. I think.